cluster-provider-client/provider.go

```diff
@@ -19,28 +19,107 @@ package clusterproviderclient
 import (
 	"context"
 
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/klog/v2"
+
 	clusterprovider "github.com/kcp-dev/edge-mc/cluster-provider-client/cluster"
 	kindprovider "github.com/kcp-dev/edge-mc/cluster-provider-client/kind"
 	v1alpha1apis "github.com/kcp-dev/edge-mc/pkg/apis/logicalcluster/v1alpha1"
+	edgeclient "github.com/kcp-dev/edge-mc/pkg/client/clientset/versioned"
 )
 
 var ProviderList map[string]ProviderClient
 
 // ES: return and error and don't panic, move push to outside the case
-func GetProviderClient(providerType v1alpha1apis.ClusterProviderType, providerName string) ProviderClient {
+func GetProviderClient(ctx context.Context,
+	clientset edgeclient.Interface,
+	providerType v1alpha1apis.ClusterProviderType,
+	providerName string) (ProviderClient, error) {
+	logger := klog.FromContext(ctx)
 	key := string(providerType) + "-" + providerName
 	newProvider, exists := ProviderList[key]
 	if !exists {
 		switch providerType {
 		case v1alpha1apis.KindProviderType:
 			newProvider = kindprovider.New(providerName)
+			// TODO: support deleting a provider from the list
 			ProviderList[key] = newProvider
+			w, _ := newProvider.Watch()
+			go func() {
+				for {
+					event, ok := <-w.ResultChan()
+					if !ok {
+						w.Stop()
+						logger.Info("stopping")
+						// TODO: return an error
+						return
+					}
+					var eventLogicalCluster v1alpha1apis.LogicalCluster
```

Comment:
Why the name is "event"  it should be the lc itself.

Reply:
The 'event' is just the event sent from the watcher.  Take a look at cluster-provider-client/kind/kind.go:133 for an example of how it's created.